cpp/src/io/parquet/experimental/hybrid_scan_helpers.cpp

```diff
@@ -90,10 +90,13 @@ aggregate_reader_metadata::aggregate_reader_metadata(cudf::host_span<uint8_t con
   num_rows          = calc_num_rows();
   num_row_groups    = calc_num_row_groups();
 
-  // Force all columns to be nullable
+  // Force all leaf columns to be nullable
   auto& schema = per_file_metadata.front().schema;
   std::for_each(schema.begin(), schema.end(), [](auto& col) {
-    col.repetition_type = FieldRepetitionType::OPTIONAL;
+    // Modifying the repetition type of lists converts them to structs, so we must skip that
```

Comment:
Forcing the `col.repetition_type == OPTIONAL` for `list`s converts them to `struct`s. Make sure only to set this for leaf columns